flutter.el

```diff
@@ -129,13 +129,23 @@ ARGS is a space-delimited string of CLI flags passed to
    (let ((proc (get-buffer-process flutter-buffer-name)))
      (comint-send-string proc command))))
 
-(defun flutter--test (&optional args)
+(defun flutter--test (&rest args)
   "Execute `flutter test` inside Emacs.
 
-ARGS is a space-delimited string of CLI flags passed to
+ARGS is a list of CLI flags passed to
 `flutter`, and can be nil."
   (flutter--from-project-root
-   (compilation-start (format "%s test %s" (flutter-build-command) (or args "")) t)))
+   (compilation-start (format "%s test %s" (flutter-build-command) (mapconcat 'identity args " ")) t)))
+
+(defun flutter--find-test-case (line)
+  "Search backwards for test name starting at LINE on current buffer."
+  (save-excursion
+    (goto-char (point-min))
+    (forward-line (1- line))
+    (end-of-line)
+    (if (re-search-backward (concat "^[ \t]*\\(testWidgets\\|test\\|group\\)"
+                                    "(\\([\"'].*?[\"']\\),") nil t)
```

Comment:
This will give incomplete matches when there are nested quotes like `"test 'foo'"`. This case may be rare, but I think it's worth handling.

It will also fail for escaped quotes like `"test \"foo\""`, but this is probably quite rare and I am OK with not handling it. But if you wanted to do so, please see [this trick](https://stackoverflow.com/a/11819111/448068).

Reply:
Good point, but I think it works because the `,` on the end of regex, you can check it [here](https://regex101.com/r/1Fj3Ae/1)

Reply:
It doesn't work if there is a comma in the string; see the last line [here](https://regex101.com/r/1Fj3Ae/2).

Reply:
@amake, what about [this regex](https://regex101.com/r/JHsJG8/1)?

Reply:
I think you'll need [this one](https://regex101.com/r/JHsJG8/3).

Reply:
Oops, I forgot that Emacs doesn't have negative lookbehind.

Reply:
I guess [this](https://regex101.com/r/JHsJG8/5) is the workaround: two options between the quotes:

1. at least one non-`\` before we swallow the `\`s leading up to the closing quote
2. just an even number of (or zero) `\`s leading up to the closing quote

Converting that monstrosity to Emacs regexp with [`rxt-pcre-to-elisp`](https://github.com/joddie/pcre2el/blob/0b5b2a2c173aab3fd14aac6cf5e90ad3bf58fa7d/pcre2el.el#L1085) gives

```
(\(["']\)\(?:.*[^\]\(?:\\\\\)*\|\(?:\\\\\)*\)\1,
```

As a string literal this is (via pasting into `(rxt-pcre-to-elisp (read-string "regexp: "))`:

```
"(\\([\"']\\)\\(?:.*[^\\]\\(?:\\\\\\\\\\)*\\|\\(?:\\\\\\\\\\)*\\)\\1,"
```

Reply:
I just can't make it work on `elisp` :pensive: 
I'm trying this:
```elisp
"(\\([\"'].*?\\(?<!\\(?<!\\\\\\)\\\\\\)[\"']\\),"
```
But emacs says it's a invalid regexp

Reply:
It is because Emacs doesn't have negative lookbehind. Check out the other comments I left in the meantime.

Reply:
Sorry, I didn't refresh the github page, I'll test your suggestions, thanks!

Reply:
Hum, but this crazy regex is matching the group of a `"` or `'` :thinking:

Reply:
That's how it ensures you end with the right kind of quote. Otherwise you need to alternate two repetitive patterns, one for `'` and one for `"`. Obviously that affects the group you need to use for the ultimate match.

(I just realized I made the important part a non-capturing group. You'll have to undo that by removing the first `?:`.)

Reply:
Ok, removing the `?:`, match the second group. So should I use a `(match-string 2)` then ?

Reply:
Oh no, it's the first match, but I need a different group, right?

Reply:
> Ok, removing the `?:`, match the second group. So should I use a `(match-string 2)` then?

This sounds right to me. Did that not work?

Reply:
No :disappointed:, it's weird that `(match-string 2)` returns nil, while `(match-string 1)` returns `"`

Reply:
I think you might have removed the wrong `?:`. Either way, I pushed the fix.